### DISCUSSION TITLE
Neotree reveal and float not working when added to keymap

### DISCUSSION THREAD

**Discussion by timjackleus** [LazyVim/LazyVim#1355] [Q&A] [answered]
I have a problem with Neotree that might not be specific to LazyVim but I'm not sure. The thing is I prefer showing the file tree in a floating window, and in order to highlight the current file the following argument needs to be passed in
```
:Neotree reveal
```
If I'm navigating around then pass this command the current file is highlighted as expected, but when I'm trying this with a keymap e.g.
```
vim.keymap.set("n", "<leader>e", "<cmd>Neotree reveal<CR>", { silent = false })
```
it's no longer working, maybe the first file but once I navigate to another file and trigger the key, the initial file is still highlighted. BUT if I bind it to something else like `<leader>ee` instead it's working...

I have tried the `:map <leader>e` to reveal what's actually binded and it tells me my binding with reveal included is being used but it's not.

So first off, anyone else with the same experience? Secondly any suggestions on how I can proceed debuging this? `<leader>e` is definitely the key I want to keep for this.

**Comment by dpetka2001** [comment]
Can you try the following
```lua
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    keys = {
      { "<leader>e", false },
    },
  },
}
```
while also preserving your keymap in `~/.config/nvim/lua/config/keymaps.lua`? Also you say you prefer to show neo-tree in a floating window, but that is not what you're doing in your keymap. I believe it should be `Neotree reveal float` in your `vim.keymap.set`.

Edit: put the code snippet in a file in `~/.config/nvim/lua/plugins/neo-tree.lua`

**Reply by timjackleus** [reply]
Yes my keymaps are located in config/keymaps.lua, `position = float` was part of my config but I removed it all now and included `float` in the keymaps instead for clarity e.g. `Neotree reveal float`.
I was not able to run the snippet you've provided without causing Lazyvim to throw errors, I did try to remove the initial keymap but it did not help
```
vim.keymap.del("n", "<leader>e")
vim.keymap.set("n", "<leader>e", "<cmd>Neotree float reveal<CR>", { silent = false })
```

**Reply by dpetka2001** [reply, marked as answer]
You  should remove the initial keymap by using the `keys` section of Neo-tree spec. Just as provided in the snippet code above. Just make a new file `~/.config/nvim/lua/plugins/neo-tree.lua` and put the contents of the snippet inside. And then add your `vim.keymap.set` in `keymaps.lua`. I tested it and it works. You must be doing something wrong on your end.

Edit: and the more correct way would be to just overwrite the default keymap in the `keys` section with the command you want to run. Setting a `vim.keymap.set` is just an extra step that you don't have to do in my opinion. But you can still do it if you like it that way.

**Reply by timjackleus** [reply]
Turns out the error was thrown from my keymap.lua due to the `<leader>e` mapping not existing anymore. But if I changed the key directly in the keys object as you suggested it works like a charm!
e.g. this was my solution:

```
  {
    "nvim-neo-tree/neo-tree.nvim",
    keys = {
      { "<leader>e", "<cmd>Neotree reveal float<cr>" },
    },
...
```

Thanks a ton @dpetka2001 ! 💯
